### PR TITLE
bf: ARSN-57 log correct client ip

### PR DESCRIPTION
**Pull request by naren-rajendran**
Log correct client ip.

### CODE REVIEW

File: lib/s3routes/routes.js [hunk 1]
```diff
@@ -134,9 +136,10 @@ function checkTypes(req, res, params, logger) {
  *  values for whether queries are supported
  * @param {function} params.dataRetrievalFn - function to retrieve data
  * @param {RequestLogger} logger - werelogs logger instance
+ * @param {String} [s3config] - s3 configuration
```
Comment (miniscruff) [inline lib/s3routes/routes.js:143]: Is this actually optional? It looks required to me.

File: lib/s3routes/routes.js [hunk 2]
```diff
@@ -134,9 +136,10 @@ function checkTypes(req, res, params, logger) {
...
  * @returns {undefined}
  */
-function routes(req, res, params, logger) {
+function routes(req, res, params, logger, s3config) {
     checkTypes(req, res, params, logger);
```
Comment (miniscruff) [inline lib/s3routes/routes.js:147]: Should you add the new value to the check types method?

**Review by miniscruff — CHANGES_REQUESTED** [review]
Should there be a test as well?

File: lib/s3routes/routes.js (hunk 1 above)
Comment (naren-rajendran) [reply lib/s3routes/routes.js:143]: Yes optional. Not sure if this being used elsewhere.

File: lib/s3routes/routes.js (hunk 2 above)
Comment (naren-rajendran) [reply lib/s3routes/routes.js:143]: It is optional, I will add type check if it is required.

**Comment by naren-rajendran** [comment]
> Should there be a test as well?

Unit tests are already available for the function being used. I can add functional test only when arsenal is merged and the updated version is used in cloudserver.

File: lib/s3routes/routes.js (hunk 2 above)
Comment (naren-rajendran) [reply lib/s3routes/routes.js:147]: I added a check now.

**Comment by naren-rajendran** [comment]
@bert-e reset

**Review by nicolas2bert — APPROVED** [review]

**Review by miniscruff — APPROVED** [review]

**Review by miniscruff — APPROVED** [review]

**Review by jonathan-gramain — APPROVED** [review]
LGTM, but I would have expected this fix to be targeting 7.4, since it's still our current LTS and in active maintenance.

**Review by alexanderchan-scality — APPROVED** [review]

**Comment by naren-rajendran** [comment]
Opening a new PR with development/7.4 as base.

**Pull request closed without merging**